### PR TITLE
Conversation Activities from Tab- C# with Razor

### PR DESCRIPTION
**Pull request by v-kalc**
The sample is to Create conversational tabs with the following actions:

- Create a new conversation.
- Close conversation view.
- Continue an existing conversation.
- Deeplink to redirect to a conversation.

### CODE REVIEW

**Review by v-chetsh — CHANGES_REQUESTED** [review]
Added my comments, please do changes accordingly.

**Review by v-chetsh — APPROVED** [review]
All looks good, approving the PR.

File: README.md [hunk 1]
```diff
@@ -26,6 +26,7 @@
 |11|Integrate graph toolkit component in teams tab      | Microsoft Teams tab sample app for demonstrating graph toolkit component |                      | [View][tab-graph-toolkit#js]|
 |12|Device permissions      | Microsoft Teams tab sample app for demonstrating device permissions |                      | [View][tab-device-permissions#js]|
 |13|Show adaptive cards in Teams tab | Microsoft Teams tab sample code which demonstrates how to [show adaptive cards in Teams tab](https://docs.microsoft.com/microsoftteams/platform/tabs/how-to/build-adaptive-card-tabs) | [View][tab-adaptive-cards#csharp]               |                | [View][tab-adaptive-cards#js]|
+|14|Create Conversational tab     | Microsoft Teams tab sample app for demonstrating create conversation tab | [View][tab-conversation#csharp] |           |
```
[the remark this answers is not in the document]
Comment (v-kalc) [reply README.md:29]: Updated

File: samples/tab-conversations/csharp/TabConversation/Pages/configure.cshtml [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+        var tabChoice = document.getElementById('tabChoice');
+        var selectedTab = tabChoice[tabChoice.selectedIndex].value;
+
+        return (
+            window.location.protocol +
+            '//' +
+            window.location.host +
+            '/' +
+            selectedTab
+        );
```
[the remark this answers is not in the document]
Comment (v-kalc) [reply samples/tab-conversations/csharp/TabConversation/Pages/configure.cshtml:57]: removed

File: samples/tab-conversations/csharp/TabConversation/Properties/launchSettings.json [hunk 3]
```diff
@@ -0,0 +1,29 @@
+{
```
[the remark this answers is not in the document]
Comment (v-kalc) [reply samples/tab-conversations/csharp/TabConversation/Properties/launchSettings.json:1]: we are defining fixed local porting address here

File: samples/tab-conversations/csharp/TabConversation/wwwroot/js/site.js [hunk 4]
```diff
@@ -0,0 +1,4 @@
+﻿// Please see documentation at https://docs.microsoft.com/aspnet/core/client-side/bundling-and-minification
```
[the remark this answers is not in the document]
Comment (v-kalc) [reply samples/tab-conversations/csharp/TabConversation/wwwroot/js/site.js]: removed

File: samples/tab-conversations/csharp/TabConversation/Pages/conversation-tab.cshtml [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+                        "subEntityId": "task-2037",
+                        "entityId": context.entityId,
+                        "channelId": context.channelId,
+                        "title": "Task Title"
+                    })
+                microsoftTeams.conversations.onStartConversation = (conversationReponse) => {
+                    console.log(conversationReponse)
+                };
+            })
+
```
[the remark this answers is not in the document]
Comment (v-kalc) [reply samples/tab-conversations/csharp/TabConversation/Pages/conversation-tab.cshtml]: removed

File: samples/tab-conversations/csharp/TabConversation/Pages/configure.cshtml [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+                // This API tells Microsoft Teams to enable the 'Save' button. Since Microsoft Teams always assumes
+                // an initial invalid state, without this call the 'Save' button will never be enabled.
+                microsoftTeams.settings.setValidityState(
+                    selectedTab === 'conversationTab' || selectedTab === 'conversationTab'
+                );
+            }
+        }
+
+        // Create the URL that Microsoft Teams will load in the tab. You can compose any URL even with query strings.
+    function createTabUrl() {
```
[the remark this answers is not in the document]
Comment (v-kalc) [reply samples/tab-conversations/csharp/TabConversation/Pages/configure.cshtml]: Aligned

File: samples/tab-conversations/csharp/TabConversation/Pages/Index.cshtml.cs [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+
+        public IndexModel(ILogger<IndexModel> logger)
+        {
+            _logger = logger;
+        }
+
+        public void OnGet()
+        {
+
+        }
```
[the remark this answers is not in the document]
Comment (v-kalc) [reply samples/tab-conversations/csharp/TabConversation/Pages/Index.cshtml.cs]: removed

**Review by Wajeed-msft — APPROVED** [review]

**Pull request merged**